packages/core/admin/admin/src/content-manager/pages/ListView/components/ViewSettingsMenu/tests/index.test.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+
+import { lightTheme, ThemeProvider } from '@strapi/design-system';
+import { fireEvent, render as renderRTL } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import { IntlProvider } from 'react-intl';
+import { Provider } from 'react-redux';
+import { Router } from 'react-router-dom';
+import { combineReducers, createStore } from 'redux';
+
+import reducers from '../../../../../../reducers';
+import { ViewSettingsMenu } from '../index';
+
+const layout = {
+  contentType: {
+    attributes: {
+      id: { type: 'integer' },
+      name: { type: 'string' },
+      createdAt: { type: 'datetime' },
+      updatedAt: { type: 'datetime' },
+    },
+    metadatas: {
+      id: {
+        list: { label: 'id', searchable: true, sortable: true },
+      },
+      name: {
+        list: { label: 'name', searchable: true, sortable: true },
+      },
+      createdAt: {
+        list: { label: 'createdAt', searchable: true, sortable: true },
+      },
+      updatedAt: {
+        list: { label: 'updatedAt', searchable: true, sortable: true },
+      },
+    },
+    layouts: {
+      list: [],
+    },
+    options: {},
+    settings: {},
+  },
+};
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  // eslint-disable-next-line react/prop-types
+  CheckPermissions: ({ children }) => <div>{children}</div>,
+}));
+
+const history = createMemoryHistory();
+history.push = jest.fn();
```

Comment:
Why do you have to do that? Could you please inline the router creation in `wrapper`?

Reply:
I need the history variable in scope with this unit test where I test the redirect button, https://github.com/strapi/strapi/blob/d39ff305a75c58512924149e9f3ab7c336d55107/packages/core/admin/admin/src/content-manager/pages/ListView/components/ViewSettingsMenu/tests/index.test.js#L126 If I move the history definition inside the wrapper I have errors

Reply:
Mh, I think this is not the best way to handle this, because you are testing internals. I'd remove the `push` mock and actually test the pathname like: `expect(history.location.pathname).toBe('/your/path');`. I'm not 100% sure the attribute is called pathname, but I think you know what I mean :)